### ISSUE TITLE
Prep codebase for public release

### ISSUE THREAD

**Issue by yabberyabber** [Team973/2018-inseason#132]
The following things need to be done before we can make this repo public:
 - add javadocs to important methods
 - merge into master
 - Add a section to readme.md explaining how the codebase is laid out (brief summary of what directories contain what)

Should we make a post to chief? I think there's a lot to show off in here but I don't like being showy
